pkg/webui/console/views/device-authentication-code/index.js

```diff
@@ -0,0 +1,212 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import * as Yup from 'yup'
+import { connect } from 'react-redux'
+import { Col, Row, Container } from 'react-grid-system'
+import { defineMessages } from 'react-intl'
+
+import Form from '../../../components/form'
+import Input from '../../../components/input'
+import SubmitBar from '../../../components/submit-bar'
+import SubmitButton from '../../../components/submit-button'
+import IntlHelmet from '../../../lib/components/intl-helmet'
+import ModalButton from '../../../components/button/modal-button'
+import Notification from '../../../components/notification'
+import toast from '../../../components/toast'
+
+import { updateDevice } from '../../store/actions/devices'
+import { attachPromise } from '../../store/actions/lib'
+import { selectSelectedApplicationId } from '../../store/selectors/applications'
+import { selectSelectedDevice, selectSelectedDeviceId } from '../../store/selectors/devices'
+
+import sharedMessages from '../../../lib/shared-messages'
+import PropTypes from '../../../lib/prop-types'
+
+const m = defineMessages({
+  deleteCode: 'Delete authentication code',
+  deleteWarning: 'Are you sure you want to delete this authentication code?',
+  noCodeSet: 'There is currently no authentication code set',
+  validateToDate: 'Invalid code expiration date',
+  deleteFailure: 'There was a problem deleting the authentication code',
+  deleteSuccess: 'The authentication code has been deleted successfully',
+  updateSuccess: 'The authentication code has been updated successfully',
+})
+
+const validationSchema = Yup.object({
+  claim_authentication_code: Yup.object({
+    value: Yup.string()
+      .matches(/^[A-Z0-9]{1,32}$/, sharedMessages.validateFormat)
```

Comment:
The validation message should be clearer, eg. telling that only uppercase letters and numbers are allowed.